File: app/src/main/java/com/pulltorefresh/app/MainActivity.java
package com.pulltorefresh.app;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.ImageView;

public class MainActivity extends AppCompatActivity {
    private ImageView refreshAnim;//点点动画的ImageView
    private MoveImageView moveImageView;
    private MoveImageHelper moveImageHelper;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        refreshAnim = (ImageView) findViewById(R.id.iv_pull_to_refresh_anim);
        moveImageView = (MoveImageView) findViewById(R.id.iv_pull_to_refresh_bg);
        moveImageHelper = new MoveImageHelper(refreshAnim, moveImageView);
        moveImageHelper.start();
    }
}

File: app/src/main/java/com/pulltorefresh/app/MainActivity.kt
package com.pulltorefresh.app

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.ImageView

class MainActivity : AppCompatActivity() {
    private val moveImageView:MoveImageView by lazy {
        findViewById<MoveImageView>(R.id.iv_pull_to_refresh_bg)
    }
    private val refreshAnimIv:ImageView by lazy {
        findViewById<ImageView>(R.id.iv_pull_to_refresh_anim)
    }
    private lateinit var moveImageHelper:MoveImageHelper
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        moveImageHelper = MoveImageHelper(refreshAnimIv!!,moveImageView!!)
        moveImageHelper.start()
    }
}
